src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -62,8 +74,22 @@ internal void Match(RouteContext context)
                 }
             }
 
-            context.Parameters = parameters;
-            context.Handler = Handler;
+            // We track the number of segments in the template that matched
+            // against this particular route then only select the route that
+            // matches the most number of segments on the route that was passed.
+            // This check is an exactness check that favors the more precise of
+            // two templates in the event that the following route table exists.
+            //  Route 1: /{anythingGoes}
+            //  Route 2: /users/{id:int}
+            // And the provided route is `/users/1`. We want to choose Route 2
+            // over Route 1.
+            var lengthNonEmptySegments = context.Segments.Where(t => t != string.Empty).Count();
```

Comment:
An addendum to this comment, if the only route entry is `/[anythingGoes}` and the route is `/users/1` then a match will not occur. I assume this is what we want...

Reply:
That's right. In server-side ASP.NET Core, there's a separate syntax for capturing values that span multiple segments - [see "catch-all" parameters](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-3.1#route-template-reference).

I'm not suggesting we need support for catch-all parameters right now, just that I agree with what you're doing here because we wouldn't want `/{myparam}` to match `/a/b/c/d` as that would force us into a making a breaking change in the future if we do add support for catch-all parameters.